### ISSUE TITLE
'sensor.yambms_yambms_1_battery_capacity_s' has "unknown state"

### ISSUE THREAD

**Issue by vl-pub** [Sleeper85/esphome-yambms#14]
hello

'sensor.yambms_yambms_1_battery_capacity_s' has "unknown" state"- see screenshot

I can't find such sensor in the board code (board - ESP32-S3)

Could you please check?

Thanks Val.
![yambms_yambms_1_battery_capacity_s](https://github.com/user-attachments/assets/928f1b8c-13f7-4b64-b304-3f535b7e1d9b)

**Comment by Sleeper85** [comment]
How many BMS are combined ?

**Comment by vl-pub** [comment]
Only one

**Comment by vl-pub** [comment]
Hmmm...
It's interesting....

After switching "Combine" control On/Off several times something updated and "Battery Capacity" displays correct value:

![image](https://github.com/user-attachments/assets/a0f3c16a-1b51-47e8-86ee-2644dcf3abb8)

Also I've found following scenario:


1. Set "Combine" to "ON"
2. Disconnect LAN cable from BMS
3. Reboot ESP board

![image](https://github.com/user-attachments/assets/13eeb52e-bb68-449b-a929-82c83aade02c)

4. Connect LAN cable to BMS - correct values are displayed:

![image](https://github.com/user-attachments/assets/deedea70-d1d7-4de7-987c-cf764da2bdea)

5.  Set "Combine" to "OFF" - "Battery SOH", "Battery capacity", "Capacity Remaining" sensors display zero values:

![image](https://github.com/user-attachments/assets/5391803f-858f-444d-92be-2989cbee2852)

6. Set "Combine" to "ON" - - correct values are displayed again:

![image](https://github.com/user-attachments/assets/57cf6482-d3bb-42a5-8bad-07c8fbc4d717)

7.  Set "Combine" to "OFF"
8. Reboot ESP board

9. Nothing changed:

![image](https://github.com/user-attachments/assets/2522684b-bce6-4698-8294-514200cba77d)

**Comment by vl-pub** [comment]
Seems "Combine" must be set to "ON" value for one connected BMS 
I'll check dashboard tomorrow morning one more time because of issue from topic has reproduced after some time after reboot.

thanks

**Comment by vl-pub** [comment]
hello
here is update.

after several hors and switching between YamBMS/Control/BMS tabs "Battery Capacity" sensor lost value and has status "Unknown" ( "Combine" is set to "ON"):

![image](https://github.com/user-attachments/assets/c91ff0e5-79b0-4a53-8a32-f47045ae1e6d)

"Combine" value has been changed between "ON"/"OFF" sever times - nothing changed

ESP board has been restarted, and "Combine" value has been changed between "ON"/"OFF" sever times again  - 

"Battery Capacity" sensor became available and has correct value:

![image](https://github.com/user-attachments/assets/6a92878c-bcf6-4d68-877f-7394fba41324)


So, some problem is still exist.
Could you please check from your side?
I'm ready to provide needed information / perform required testing.

Thank you,
Val.

**Comment by Sleeper85** [comment]
Can you send me your YAML please?

**Comment by vl-pub** [comment]
What yaml? Dashboard Yaml?

**Comment by Sleeper85** [comment]
No, your main YAML.

**Comment by vl-pub** [comment]
attached

[one-bms_JK-PB_RS485_mode2_MY.yaml.txt](https://github.com/user-attachments/files/18246330/one-bms_JK-PB_RS485_mode2_MY.yaml.txt)

**Comment by Sleeper85** [comment]
> Seems "Combine" must be set to "ON" value for one connected BMS

An uncombined BMS is an unused BMS, all this is explained in the doc. You can uncombine it with the `Combine` switch and it can be uncombined automatically if the combination conditions are no longer respected.

Everything you have tested and shown in the picture is normal. If you have no combined BMS the values ​​are at 0% and the information is no longer sent on the CAN bus.

The only thing that is not normal is to have an `Unknown` value.

**Installed Battery Capacity:** this value is updated when a BMS is combined for the first time, then the value will remain unchanged if the BMS is uncombined, this corresponds to your total capacity regardless if the BMS are combined or not.

**Battery Capacity:** this is the sum of the battery capacity of all BMS currently combined. This value is updated when a BMS is combined or uncombined.

**Remainnig Capacity:** the remaining capacity of all BMS currently combined. This value is updated continuously.

**Comment by vl-pub** [comment]
thank you for answer!

I think this issue can be closed now or after some time.
I'll inform you if "The only thing that is not normal is to have an Unknown value." issue occurred again and I'll have clear scenario.

Thanks again!
Val

**Comment by Sleeper85** [comment]
Ok, I'll leave this issue open, keep me informed.

**Comment by Sleeper85** [comment]
FYI

https://github.com/esphome/esphome/pull/7547

**Comment by Sleeper85** [comment]
In YamBMS 1.5.3 I changed the condition to combine battery capacity info.

Can you tell me if it works better for you?

**Comment by vl-pub** [comment]
is it ok that versions are differ??

![Image](https://github.com/user-attachments/assets/823372b5-8087-431d-a9a9-a29fa78c3808)

**Comment by Sleeper85** [comment]
Yes this is normal, I leave a few days to make corrections if necessary before creating the version tag.

**Comment by vl-pub** [comment]
Hmmm...

It's a little bit worse now.

1. I unplug RS485 cable from BMS 
2. All sensors are in Unavailable state
3. Plug cable again
4. All sensors restored except "Battery Capacity" sensor
5. ESP32 reboot is required to restore normal work

**Comment by Sleeper85** [comment]
And the `Installed Battery Capacity` too ?

**Comment by vl-pub** [comment]
1. BMS unplugged

![Image](https://github.com/user-attachments/assets/dafc36e5-afdd-473f-8b0e-2be29ce975bf)

2. BMS plugged again

![Image](https://github.com/user-attachments/assets/bc7ae9a4-b551-4ea1-b67f-4f621ed895af)

**Comment by Sleeper85** [comment]
Ok I check it this afternoon.

**Comment by Sleeper85** [comment]
Fixed.

Copy the new packages folder and recompile to test it.

**Comment by vl-pub** [comment]
Looks like issue is fixed
Tested 2 scenarios mentioned below

Thanks
Val

-----------------------------------


**Scenario1**

1. Unplug BMS cable

![Image](https://github.com/user-attachments/assets/2f37c99f-9147-44ac-8d6d-733a3560d84a)

2. Plug BMS cable

![Image](https://github.com/user-attachments/assets/244ee241-8825-4561-b082-8033836e5404)


**Scenario 2**

1. Unplug BMS cable
2. Reboot ESP32

![Image](https://github.com/user-attachments/assets/fa3dd074-0b90-4ebd-8e87-d2bb6564abd1)

3. Plug BMS cable

![Image](https://github.com/user-attachments/assets/6b704bcb-09ab-462f-b926-b2759915c9be)

**Comment by Sleeper85** [comment]
I just made another correction => [Don't count a uncombined BMS as blocking charge/discharge](https://github.com/Sleeper85/esphome-yambms/commit/91df96d70cdfe5b9f13691cc72330e9b5c1dcecd)

This is unrelated to this issue.
